MonstieWash/Assets/Scripts/Player/ItemPickup.cs

```diff
@@ -0,0 +1,248 @@
+                closestObjDistFromCentre = Vector3.Distance(transform.position, closestObj.transform.position);
+            }
+        }
+
+        return closestObj.GetComponent<StuckItem>();
+    }
+
+    /// <summary>
+    /// Lets go of a held item.
+    /// </summary>
+    private void LetGoItem()
+    {
+        m_heldItem = null;
+    }
+
+    /// <summary>
+    /// Picks up the given item.
+    /// </summary>
+    /// <param name="item">The item to pick up.</param>
+    private void PickupItem()
+    {
+        m_itemInitialParent = m_heldItem.transform.parent;
+        m_heldItem.transform.parent = heldItemsTransform;
+        m_heldItem.transform.localPosition = Vector3.zero; // snap to centre of hand
+        m_heldItem.Rb.angularVelocity = 0f;
+    }
+
+    /// <summary>
+    /// Drops the held item.
+    /// </summary>
+    private void DropItem()
+    {
+        m_heldItem.transform.parent = m_itemInitialParent;
+
+        var handMovement = GetComponent<PlayerHand>().Velocity.normalized;
+        handMovement *= throwSpeedMultiplier;
+
+        var rotationalVelocity = UnityEngine.Random.Range(90f, 540f) * Mathf.Sign(handMovement.x);
+        m_heldItem.Rb.velocity = handMovement;
+        m_heldItem.Rb.angularVelocity = rotationalVelocity;
+
+        m_heldItem = null;
+    }
+
+    /// <summary>
+    /// Sets up the appropriate variables for wiggling.
+    /// </summary>
+    private void SetupHoldingStuckItem()
+    {
+        m_initialRotation = m_heldItem.transform.rotation.eulerAngles.z;
+
+        SetMinimumDir();
+
+        m_prevPos = transform.position;
+    }
+
+    /// <summary>
+    /// Sets the hands position to the point on the holding circle dictated by the held item.
+    /// </summary>
+    private void SetHandPosition()
+    {
+        if (!m_holding) return;
+
+        var movement = transform.position - m_prevPos;
+        var dir = (transform.position + movement - m_heldItem.transform.position).normalized;
+
+        var relativeAngle = Vector3.SignedAngle(m_minimumDir, dir, Vector3.forward);
+        var relativeOffset = m_initialRotation - (m_heldItem.MaxRotation / 2f);
+
+        var overMaxAngle = relativeAngle > m_heldItem.MaxRotation;
+        var underMinAngle = relativeAngle < 0f;
+
+        var finalRelativeAngle = relativeAngle + relativeOffset;
+
+        if (overMaxAngle || underMinAngle)
+        {
+            Wiggled(overMaxAngle);
+
+            var baseTheta = overMaxAngle ? m_heldItem.MaxRotation : 0f;
+            var relativeTheta = baseTheta + relativeOffset;
+            var x = Mathf.Cos(relativeTheta * Mathf.Deg2Rad);
+            var y = Mathf.Sin(relativeTheta * Mathf.Deg2Rad);
+
+            dir = new Vector3(x, y, 0f).normalized;
+            finalRelativeAngle = relativeTheta;
+        }
+
+        transform.position = m_heldItem.transform.position + dir * m_heldItem.GrabDistance;
+        m_heldItem.transform.rotation = Quaternion.Euler(0f, 0f, finalRelativeAngle);
```

Comment:
Might not be this specific line, but this bug seems to be caused by rotations so try looking here first. Currently a bug exists where if the player is holding a stuck item (attempting to wiggle the bone) and switches tools while still holding the bone, the bone begins to rotate in a 360 degree spin anchored on the wiggle point.